monitor_ingest/actions/setup_entity.py

```diff
@@ -0,0 +1,99 @@
+import yaml
+from st2common.runners.base_action import Action
+from mam.sdk import (entitytype, constants, kpifunction, dimension)
+
+__all__ = ['SetupEntityAction']
+
+
+class SetupEntityAction(Action):
+    def __init__(self, config):
+        super(SetupEntityAction, self).__init__(config)
+        self._config = self.config
+        self._credentials = self._config.get('credentials', None)
+        self._data_file_path = self._config.get('data_file_path', None)
+        self._action_type = self._config.get('action_type', None)
+
+        if not self._config:
+            raise ValueError('Missing config yaml')
+        if not self._credentials:
+            raise ValueError('Missing IBM Watson IoT credentials in config file')
+        if not self._data_file_path:
+            raise ValueError('Missing data file path in config file')
+        if not self._action_type:
+            raise ValueError('Missing action type key in config file')
+
+    def run(self):
+        operations_completed = {}
+
+        if self._action_type == "SetupEntityAction":
+            self.setup_EntityAction(operations_completed)
+        elif self._action_type == "SetupAddConstants":
+            self.setup_AddConstants(operations_completed)
+        elif self._action_type == "SetupAddDimesions":
+            self.setup_AddDimesions(operations_completed)
+        elif self._action_type == "SetupAddFunctions":
+            self.setup_AddFunctions(operations_completed)
+
+        """----------STATUS----------"""
+        self.logger.info('RESULT :')
+        for name, status in operations_completed.items():
```

Comment:
All of the logic in this loop is overcomplicated. Why not just return a tuple of `(success, result_text)` to StackStorm? Seems much cleaner to me.

Reply:
simplified the logic to return boolean values